src/rust/dbv2/migrations/2023-10-03-070930_add_order_order/up.sql

```diff
@@ -0,0 +1,25 @@
+-- Your SQL goes here
+DROP VIEW api.limit_orders;
+
+ALTER TABLE aggregator.user_history_limit ADD COLUMN prev NUMERIC;
+
+CREATE VIEW api.limit_orders AS
+    SELECT * FROM aggregator.user_history_limit NATURAL JOIN aggregator.user_history;
+
+GRANT SELECT ON api.limit_orders TO web_anon;
+
+CREATE FUNCTION linked_list_last(NUMERIC(20), NUMERIC(20)) RETURNS NUMERIC AS $$
```

Comment:
Do we necessarily need the linked list data structure in the database? I don't think so.
We should be able to select orders in the open state, ordered by price and THEN time, where market_id equals given. This is the limit order book state for the given market.

Reply:
This would work if only we did not move the order at the end of the queue when changing size to a higher amount. But you give me an idea. What about adding a `last_time_size_went_up` field (TODO: find a better name), and sort by `COALESCE(last_time_size_went_up, creation_time)` (returns first if not null, second else). @alnoki would this work ? This would greatly simplify the logic and remove room for error.

Reply:
Since all transactions in the same block have the same timestamp, you have to use (transaction version, event index) instead of a traditional timestamp. This is because multiple orders could be created and resized inside of one transaction.

I guess we could include the creation "time" but I was thinking we could get by with just `last_increase_txn/idx` which is initialized to the txn/idx of the creation event and updated with change size increases.